pkg/execution/state/redis_state/redis_state.go

```diff
@@ -322,11 +312,14 @@ func (m mgr) UpdateMetadata(ctx context.Context, runID ulid.ULID, md state.Metad
 	if !md.StartedAt.IsZero() {
 		input[2] = strconv.FormatInt(md.StartedAt.UnixMilli(), 10)
 	}
-	status, err := scripts["updateMetadata"].Exec(
+
+	fnRunState := m.s.FunctionRunState()
+
+	status, err := retriableScripts["updateMetadata"].Exec(
 		ctx,
-		m.r,
+		fnRunState.Client(runID),
```

Comment:
Not only do we construct old or new keys based on the Run ID, we also pick the sharded (feature-specific) or unsharded client to execute the operation on.